src/nsrt_learning/strips_learning/gen_to_spec_learner.py

```diff
@@ -269,17 +270,22 @@ def _backchain_one_pass(
                     for nec_pnad in param_opt_to_nec_pnads[option.parent]:
                         pre = self._induce_preconditions_via_intersection(
                             nec_pnad)
+                        self._recompute_datastores_from_segments(
```

Comment:
why do we need this...? didn't we just call recompute right above?

Reply:
lol you're so right - great catch! Removed